services/datamanager/builtin/builtin.go

```diff
@@ -365,9 +365,10 @@ func (svc *builtIn) Sync(_ context.Context) error {
 
 func (svc *builtIn) syncDataCaptureFiles() error {
 	svc.lock.Lock()
-	defer svc.lock.Unlock()
 	oldFiles := make([]string, 0, len(svc.collectors))
-	for _, collector := range svc.collectors {
+	currCollectors := svc.collectors
```

Comment:
Ah that's a good point. Yeah so if svc.collectors has collectors removed while this is running, it may not ever call sync on those old collector's files, and if some are added I think the behavior would be non-deterministic. Will open a new PR copying the map